### PR TITLE
[SPARK-36896][PYTHON] Return boolean for `dropTempView` and `dropGlobalTempView`

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Currently `dropTempView` and `dropGlobalTempView` don't have return value, which conflicts with their docstring:
`Returns true if this view is dropped successfully, false otherwise.`. And that's not consistent with the same API in other languages.

The PR proposes a fix for that.

### Why are the changes needed?
Be consistent with API in other languages.

### Does this PR introduce _any_ user-facing change?
Yes.
#### From
```py
# dropTempView
>>> spark.createDataFrame([(1, 1)]).createTempView("my_table")
>>> spark.table("my_table").collect()
[Row(_1=1, _2=1)]  
>>> spark.catalog.dropTempView("my_table")
>>> spark.catalog.dropTempView("my_table")

# dropGlobalTempView
>>> spark.createDataFrame([(1, 1)]).createGlobalTempView("my_table")
>>> spark.table("global_temp.my_table").collect()
[Row(_1=1, _2=1)]
>>> spark.catalog.dropGlobalTempView("my_table")
>>> spark.catalog.dropGlobalTempView("my_table")
```

#### To
```py
# dropTempView
>>> spark.createDataFrame([(1, 1)]).createTempView("my_table")
>>> spark.table("my_table").collect()
[Row(_1=1, _2=1)]  
>>> spark.catalog.dropTempView("my_table")
True
>>> spark.catalog.dropTempView("my_table")
False

# dropGlobalTempView
>>> spark.createDataFrame([(1, 1)]).createGlobalTempView("my_table")
>>> spark.table("global_temp.my_table").collect()
[Row(_1=1, _2=1)]
>>> spark.catalog.dropGlobalTempView("my_table")
True
>>> spark.catalog.dropGlobalTempView("my_table")
False
```

### How was this patch tested?
Existing tests.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Looks good if tests pass.

File: python/pyspark/sql/catalog.py
Comment (dongjoon-hyun) [inline python/pyspark/sql/catalog.py:53]: Shall we mention `setCurrentDatabase` together?

**Review by dongjoon-hyun — COMMENTED** [review]
Adding `boolean` return is not a breaking change. However, removing `return` statement could be a breaking API change. I'm not sure if we can remove it.
```
setCurrentDatabase(self, dbName):
```

I'll leave this to your decision, @HyukjinKwon and @xinrong-databricks . (Also, cc @ueshin )

File: python/pyspark/sql/catalog.py
Comment (HyukjinKwon) [reply python/pyspark/sql/catalog.py:53]: if I am not mistaken, previously it return `None` too because the JVM returns `null`:

```python
>>> print(spark.catalog.setCurrentDatabase("default"))
None
```

so it should be fine.

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/sql/catalog.py
Comment (HyukjinKwon) [reply python/pyspark/sql/catalog.py:53]: Let me merge this in but please let me know if there's something I missed.

Comment (dongjoon-hyun) [reply python/pyspark/sql/catalog.py:53]: Ah, got it. It makes sense.

**Pull request closed without merging**
